### ISSUE TITLE
Integer.parseUnsignedInt is not implemented

### ISSUE THREAD

**Issue by ochrons** [scala-js/scala-js#1583]
Makes it hard to parse 32-bit hex strings. Especially since Long parsing has a bug too (#1582).

**Comment by InTheNow** [comment]
Method only since Java 8 => perhaps reason for not being there?

**Comment by ochrons** [comment]
Don't see no reason to not support useful some Java8 lib features in Scala.js :)

**Comment by sjrd** [comment]
In theory, we can certainly add Java 8-specific methods. Keep in mind, though, that they will be available only when using a JDK 8 to *compile* the code.

My main concern is about *testing* this stuff. Currently, we don't have a story for testing JDK8-only APIs. Simply putting the tests in `test-suite/` will prevent the test suite to compile when using JDK6/7. We probably need another special directory `test-suite/src/test/require-jdk8/`, just like we have `require-sam/`, that is added to the sources only when compiling with a JDK 8.
